extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergTools.java

```diff
@@ -16,4 +24,77 @@ public static IcebergCatalogAdapter createAdapter(
+     * {@code "org.apache.iceberg.rest.RESTCatalog"} or {@code "org.apache.iceberg.aws.glue.GlueCatalog")}. Choices for
+     * {@code "type"} include {@code "hive"}, {@code "hadoop"}, {@code "rest"}, {@code "glue"}, {@code "nessie"},
+     * {@code "jdbc"}.</li>
+     * <li>{@code "uri"} - the URI of the catalog.</li>
+     * <li>{@code "io-impl"} - the Java FileIO implementation to use. Common choices are:
+     * {@code "org.apache.iceberg.aws.s3.S3FileIO"} and {@code "org.apache.iceberg.hadoop.HadoopFileIO"}</li>
+     * </ul>
+     * <p>
+     * Other common properties include:
+     * </p>
+     * <ul>
+     * <li>{@code "warehouse"} - the location of the data warehouse.</li>
+     * <li>{@code "client.region"} - the region of the AWS client.</li>
+     * <li>{@code "s3.access-key-id"} - the S3 access key for reading files.</li>
+     * <li>{@code "s3.secret-access-key"} - the S3 secret access key for reading files.</li>
+     * <li>{@code "s3.endpoint"} - the S3 endpoint to connect to.</li>
+     * </ul>
+     * <p>
+     * Additional properties for the specific catalog should also be included, such as as S3-specific properties for
+     * authentication or endpoint overriding.
+     * </p>
+     * 
+     * @param name the name of the catalog; if omitted, the catalog URI will be used to generate a name
+     * @param properties the map containing the Iceberg catalog properties to use
+     * @return the Iceberg catalog adapter
+     */
+    @SuppressWarnings("unused")
+    public static IcebergCatalogAdapter createAdapter(
+            @Nullable final String name,
+            @NotNull final Map<String, String> properties) {
+
+        // Validate the minimum required properties are set.
+        if (!properties.containsKey(CatalogProperties.CATALOG_IMPL)
+                && !properties.containsKey(CatalogUtil.ICEBERG_CATALOG_TYPE)) {
+            throw new IllegalArgumentException(String.format("Catalog type or implementation property '%s' is required",
+                    CatalogProperties.CATALOG_IMPL));
+        }
+        if (!properties.containsKey(CatalogProperties.URI)) {
+            throw new IllegalArgumentException(String.format("Catalog URI property '%s' is required",
+                    CatalogProperties.URI));
+        }
+        if (!properties.containsKey(CatalogProperties.FILE_IO_IMPL)) {
+            throw new IllegalArgumentException(String.format("File IO implementation property '%s' is required",
+                    CatalogProperties.FILE_IO_IMPL));
+        }
+
+        final String catalogUri = properties.get(CatalogProperties.URI);
+        final String catalogName = name != null ? name : "IcebergCatalog-" + catalogUri;
+        final String fileIOImpl = properties.get(CatalogProperties.FILE_IO_IMPL);
+
+        final Configuration hadoopConf = HadoopFileIO.class.getName().equals(fileIOImpl)
+                ? new Configuration()
+                : null;
```

Comment:
I think I've found other cases that need hadoopConf; either, when the catalog impl is HadoopCatalog (or catalog type is , or when the fileIO is ResolvingFiloIO and it resolves to HadoopFileIO.

I think we might actually want to expose it as a method param, so users can configure it. I was working with the HadoopCatalog and I needed to add custom properties to Configuration to get it to work. I expect the majority of times it can be `null` / `None`. I worry that if we don't expose it now, we'll need to expose it in the future and have a new overload.

Reply:
Added this as a parameter to the generic adapter creation and provide an empty config with our helpers.